keygen/src/ssl.c

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright 2018 Xaptum, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <openssl/pem.h>
+#include <openssl/ec.h>
+#include <openssl/x509.h>
+#include <sodium.h>
+#include <errno.h>
+
+#include "ssl.h"
+#include "enftun/config.h"
+
+/**
+ * public_key_encode_x962() - Encode a public key from an EVP_PKEY structure
+ *
+ * Caller must free the returned buffer
+ *
+ * Return: A new buffer containing the x9.62-encoded string
+ */
+static char *public_key_encode_x962(EVP_PKEY *pkey)
```

Comment:
Rather than take the `EVP_PKEY` for this, can't we just take the `EC_KEY` (`gen_key` still has the `ec_key` when it calls this function). That avoids having to create and free the `EC_KEY` here

Reply:
Agreed, changed.